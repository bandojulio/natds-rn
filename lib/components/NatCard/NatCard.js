import React from 'react';
import { Dimensions } from 'react-native';
import PropTypes from 'prop-types';
import { CardWrapper, CardImage, CardIcon, CardIHeader, CardFooter, IconContainer, TitleContainer, Title, Subtitle, Resume } from './NatCard.styles';

const width = Dimensions.width;

const defaultMargin = propMargin => {
  switch (propMargin) {
    case 'none': return '0px';
    case 'dense': return '8px';
    case 'normal': return '16px';
    default: return '0px';
  }
};

export const NatCard = ({
  onPress, type, title, subtitle, imageURI, resume, margin, accessibilityLabel, accessibilityHint, imageDescription,
}) => (
  <CardWrapper
    onPress={onPress}
    defaultMargin={defaultMargin(margin)}
    accessible={true}
    accessibilityLabel={accessibilityLabel}
    accessibilityHint={accessibilityHint}
  >
    <CardIHeader>
      <IconContainer>
        <CardIcon
          source={{uri: 'http://global-press.com/wp-content/uploads/2015/06/corp4.png'}}
        />
      </IconContainer>
      <TitleContainer>
        <Title>{title}</Title>
        <Subtitle>{subtitle}</Subtitle>
      </TitleContainer>
    </CardIHeader>
    <CardImage
      source={{uri: imageURI}}
      style={{width: width}}
      accessible={true}
      accessibilityLabel={imageDescription}
    />
    <CardFooter>
      <Resume>{resume}</Resume>
    </CardFooter>
  </CardWrapper>
);

NatCard.propTypes = {
  type: PropTypes.string,
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  imageURI: PropTypes.string.isRequired,
  resume: PropTypes.string.isRequired,
  margin: PropTypes.string,
  accessibilityLabel: PropTypes.string,
  accessibilityHint: PropTypes.string,
  imageDescription: PropTypes.string,
};

NatCard.defaultProps = {
  type: 'default',
  title: 'Title',
  subTitle: 'SubTitle',
};
