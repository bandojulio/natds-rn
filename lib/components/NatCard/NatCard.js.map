{"version":3,"sources":["NatCard.js"],"names":["defaultMargin","propMargin","styles","borderRadius","elevation","shadowColor","shadowOffset","height","width","shadowOpacity","shadowRadius","NatCard","onPress","title","subtitle","imageURI","resume","margin","accessibilityLabel","accessibilityHint","theme","palette","text","primary","spacing","spacingMicro","secondary","spacingSmall","spacingStandard","propTypes","PropTypes","string","isRequired","func","shape","defaultProps","subTitle","ThemeNaturaLight","type"],"mappings":"wIAAA,4B;AACA;;;AAGA,qC;AACA;AACA;AACA;AACA,+D;;;;AAIA,GAAMA,eAAgB,QAAhBA,cAAgB,CAACC,UAAD,CAAgB;AACpC,OAAQA,UAAR;AACE,IAAK,OAAL,CAAc,MAAO,GAAP;AACd,IAAK,MAAL,CAAa,MAAO,GAAP;AACb,IAAK,OAAL,CAAc,MAAO,IAAP;AACd,IAAK,UAAL,CAAiB,MAAO,IAAP;AACjB,QAAS,MAAO,GAAP,CALX;;AAOD,CARD;;AAUA,GAAMC,QAAS;AACbC,aAAc,CADD;AAEbC,UAAW,CAFE;AAGbC,YAAa,MAHA;AAIbC,aAAc;AACZC,OAAQ,CADI;AAEZC,MAAO,CAFK,CAJD;;AAQbC,cAAe,IARF;AASbC,aAAc,CATD,CAAf;;;AAYO,GAAMC,yBAAU,QAAVA,QAAU;AACrBC,OADqB,MACrBA,OADqB;AAErBC,KAFqB,MAErBA,KAFqB;AAGrBC,QAHqB,MAGrBA,QAHqB;AAIrBC,QAJqB,MAIrBA,QAJqB;AAKrBC,MALqB,MAKrBA,MALqB;AAMrBC,MANqB,MAMrBA,MANqB;AAOrBC,kBAPqB,MAOrBA,kBAPqB;AAQrBC,iBARqB,MAQrBA,iBARqB;AASrBC,KATqB,MASrBA,KATqB;;AAWrB,8BAAC,iBAAD,EAAM,MAAOlB,MAAb;AACE,8BAAC,oBAAD;AACE,QAASU,OADX;AAEE,cAAeZ,cAAciB,MAAd,CAFjB;AAGE,eAHF;AAIE,mBAAoBC,kBAJtB;AAKE,kBAAmBC,iBALrB;;AAOE,8BAAC,kBAAD;AACE,OAAQJ,QADV;AAEE,KAAK,MAFP,iDAPF;;AAWE,8BAAC,mBAAD;AACE,8BAAC,gBAAD;AACE,KAAK,IADP;AAEE,MAAOF,KAFT;AAGE,MAAOO,MAAMC,OAAN,CAAcC,IAAd,CAAmBC,OAH5B;AAIE,OAAQH,MAAMI,OAAN,CAAcC,YAJxB,iDADF;;AAOE,8BAAC,gBAAD;AACE,KAAK,SADP;AAEE,MAAOX,QAFT;AAGE,MAAOM,MAAMC,OAAN,CAAcC,IAAd,CAAmBI,SAH5B;AAIE,OAAQN,MAAMI,OAAN,CAAcG,YAJxB,iDAPF;;AAaE,8BAAC,gBAAD;AACE,KAAK,OADP;AAEE,MAAOX,MAFT;AAGE,MAAOI,MAAMC,OAAN,CAAcC,IAAd,CAAmBC,OAH5B;AAIE,OAAQH,MAAMI,OAAN,CAAcI,eAJxB,iDAbF,CAXF,CADF,CAXqB,GAAhB;;;;;;;;AAgDPjB,QAAQkB,SAAR,CAAoB;AAClBV,kBAAmBW,oBAAUC,MADX;AAElBb,mBAAoBY,oBAAUC,MAFZ;AAGlBhB,SAAUe,oBAAUC,MAAV,CAAiBC,UAHT;AAIlBf,OAAQa,oBAAUC,MAJA;AAKlBnB,QAASkB,oBAAUG,IAAV,CAAeD,UALN;AAMlBhB,OAAQc,oBAAUC,MAAV,CAAiBC,UANP;AAOlBlB,SAAUgB,oBAAUC,MAPF;AAQlBX,MAAOU,oBAAUI,KAAV,EARW;AASlBrB,MAAOiB,oBAAUC,MAAV,CAAiBC,UATN,CAApB;;;AAYArB,QAAQwB,YAAR,CAAuB;AACrBC,SAAU,UADW;AAErBhB,MAAOiB,kCAFc;AAGrBxB,MAAO,OAHc;AAIrByB,KAAM,SAJe,CAAvB","file":"NatCard.js","sourcesContent":["import React from 'react';\nimport {\n  View,\n} from 'react-native';\nimport PropTypes from 'prop-types';\nimport { NatText } from '../NatText';\nimport { NatImage } from '../NatImage';\nimport { CardWrapper, CardFooter } from './NatCard.styles';\nimport {\n  ThemeNaturaLight,\n} from '../../themes/ThemeNaturaLight';\n\nconst defaultMargin = (propMargin) => {\n  switch (propMargin) {\n    case 'micro': return '4';\n    case 'tiny': return '8';\n    case 'small': return '16';\n    case 'standard': return '24';\n    default: return '0';\n  }\n};\n\nconst styles = {\n  borderRadius: 4,\n  elevation: 1,\n  shadowColor: '#000',\n  shadowOffset: {\n    height: 1,\n    width: 0,\n  },\n  shadowOpacity: 0.24,\n  shadowRadius: 4,\n};\n\nexport const NatCard = ({\n  onPress,\n  title,\n  subtitle,\n  imageURI,\n  resume,\n  margin,\n  accessibilityLabel,\n  accessibilityHint,\n  theme,\n}) => (\n  <View style={styles}>\n    <CardWrapper\n      onPress={onPress}\n      defaultMargin={defaultMargin(margin)}\n      accessible\n      accessibilityLabel={accessibilityLabel}\n      accessibilityHint={accessibilityHint}\n    >\n      <NatImage\n        source={imageURI}\n        type=\"card\"\n      />\n      <CardFooter>\n        <NatText\n          type=\"h6\"\n          value={title}\n          color={theme.palette.text.primary}\n          margin={theme.spacing.spacingMicro}\n        />\n        <NatText\n          type=\"caption\"\n          value={subtitle}\n          color={theme.palette.text.secondary}\n          margin={theme.spacing.spacingSmall}\n        />\n        <NatText\n          type=\"body2\"\n          value={resume}\n          color={theme.palette.text.primary}\n          margin={theme.spacing.spacingStandard}\n        />\n      </CardFooter>\n    </CardWrapper>\n  </View>\n\n);\n\nNatCard.propTypes = {\n  accessibilityHint: PropTypes.string,\n  accessibilityLabel: PropTypes.string,\n  imageURI: PropTypes.string.isRequired,\n  margin: PropTypes.string,\n  onPress: PropTypes.func.isRequired,\n  resume: PropTypes.string.isRequired,\n  subtitle: PropTypes.string,\n  theme: PropTypes.shape(),\n  title: PropTypes.string.isRequired,\n};\n\nNatCard.defaultProps = {\n  subTitle: 'SubTitle',\n  theme: ThemeNaturaLight,\n  title: 'Title',\n  type: 'default',\n};\n"]}