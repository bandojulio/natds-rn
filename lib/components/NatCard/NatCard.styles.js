import styled, { css } from 'styled-components/native';
import { colors } from '../../assets/theme/colors/colors';

const inputStyles = {
  default: {
    backgroundColor: colors.primary.white,
  },
};

const getStylesFrom = styles => type => styles[type] || styles.default;

const getInputStyles = getStylesFrom(inputStyles);

export const CardWrapper = styled.TouchableOpacity`
  background-color: ${({ type }) => getInputStyles(type).backgroundColor};
  border-radius: 4px;
  margin-top: ${({ defaultMargin }) => defaultMargin || '0px'};
`;

export const CardIcon = styled.Image`
  width: 50;
  height: 50;
`;

export const CardImage = styled.Image`
  height: 120;
`;

export const CardIHeader = styled.View`
  flex-direction: row;
  padding: 10px;
`;

export const CardFooter = styled.View`
  padding: 10px;
`;

export const IconContainer = styled.View`
  flex: 1;
  padding: 5px;
`;

export const TitleContainer = styled.View`
  flex: 5;
  padding: 5px;
`;

export const Title = styled.Text`
  font-size: 14;
`;

export const Subtitle = styled.Text`
  font-size: 12;
`;

export const Resume = styled.Text`
  font-size: 12;
`;
