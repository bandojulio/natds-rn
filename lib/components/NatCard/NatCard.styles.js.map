{"version":3,"sources":["NatCard.styles.js"],"names":["inputStyles","default","backgroundColor","ThemeNaturaLight","palette","background","paper","getStylesFrom","styles","type","getInputStyles","CardWrapper","styled","TouchableOpacity","defaultMargin","CardFooter","View"],"mappings":"+YAAA,gD;AACA,+D;;AAEA,GAAMA,aAAc;AAClBC,QAAS;AACPC,gBAAiBC,mCAAiBC,OAAjB,CAAyBC,UAAzB,CAAoCC,KAD9C,CADS,CAApB;;;;AAMA,GAAMC,eAAgB,QAAhBA,cAAgB,eAAU,sBAAQC,QAAOC,IAAP,GAAgBD,OAAOP,OAA/B,EAAV,EAAtB;;AAEA,GAAMS,gBAAiBH,cAAcP,WAAd,CAAvB;;AAEO,GAAMW,iCAAcC,iBAAOC,gBAArB;AACS,kBAAGJ,KAAH,MAAGA,IAAH,OAAcC,gBAAeD,IAAf,EAAqBP,eAAnC,EADT;;AAGM,mBAAGY,cAAH,OAAGA,aAAH,OAAuBA,gBAAiB,KAAxC,EAHN,CAAN;;;AAMA,GAAMC,+BAAaH,iBAAOI,IAApB,kBAAN","file":"NatCard.styles.js","sourcesContent":["import styled from 'styled-components/native';\nimport { ThemeNaturaLight } from '../../themes/ThemeNaturaLight';\n\nconst inputStyles = {\n  default: {\n    backgroundColor: ThemeNaturaLight.palette.background.paper,\n  },\n};\n\nconst getStylesFrom = styles => type => styles[type] || styles.default;\n\nconst getInputStyles = getStylesFrom(inputStyles);\n\nexport const CardWrapper = styled.TouchableOpacity`\n  background-color: ${({ type }) => getInputStyles(type).backgroundColor};\n  border-radius: 4px;\n  margin-bottom: ${({ defaultMargin }) => defaultMargin || '0px'};\n`;\n\nexport const CardFooter = styled.View`\n  padding: 10px;\n`;\n"]}