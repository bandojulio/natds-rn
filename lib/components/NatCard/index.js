Object.defineProperty(exports,"__esModule",{value:true});var _NatCard=require('./NatCard');Object.defineProperty(exports,'NatCard',{enumerable:true,get:function get(){return _NatCard.NatCard;}});
//# sourceMappingURL=index.js.map