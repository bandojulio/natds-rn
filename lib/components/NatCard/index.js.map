{"version":3,"sources":["index.js"],"names":["NatCard"],"mappings":"uLAASA,O","file":"index.js","sourcesContent":["export { NatCard } from './NatCard';\n"]}