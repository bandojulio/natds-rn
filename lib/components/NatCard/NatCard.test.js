import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components/native';

import { NatCard } from './NatCard';

describe('TextInput component default', () => {
  it('Should render textInput component ', () => {
    const button = renderer.create(
      <NatCard
        title="teste"
        onPress={jest.fn()}
        imageURI="https://www.harley-davidson.com/content/dam/h-d/images/motorcycles/my19/softail/fxdr-114/overview/hdi/19-softail-fxdr114-hdi-hero.jpg"
        resume="teste"
      />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

});
