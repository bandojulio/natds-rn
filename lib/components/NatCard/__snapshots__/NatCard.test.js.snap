// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component default Should render textInput component  1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "marginTop": 0,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "http://global-press.com/wp-content/uploads/2015/06/corp4.png",
          }
        }
        style={
          Array [
            Object {
              "height": 50,
              "width": 50,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 5,
            "flexShrink": 1,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        teste
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 12,
            },
          ]
        }
      />
    </View>
  </View>
  <Image
    accessible={true}
    source={
      Object {
        "uri": "https://www.harley-davidson.com/content/dam/h-d/images/motorcycles/my19/softail/fxdr-114/overview/hdi/19-softail-fxdr114-hdi-hero.jpg",
      }
    }
    style={
      Array [
        Object {
          "height": 120,
        },
        Object {
          "width": undefined,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 12,
          },
        ]
      }
    >
      teste
    </Text>
  </View>
</View>
`;
